src/components/ListviewSort.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as dijitRegistry from "dijit/registry";
+import * as classNames from "classnames";
+import * as dojoLang from "dojo/_base/lang";
+import * as dojoConnect from "dojo/_base/connect";
+
+import { Dropdown, DropdownProps } from "./Dropdown";
+import { ValidateConfigs } from "./ValidateConfigs";
+import { CommonProps, ListView, ListviewSortProps, ListviewSortState, parseStyle } from "../utils/ContainerUtils";
+import "../ui/ListviewSort.css";
+
+export default class ListviewSort extends Component<ListviewSortProps, ListviewSortState> {
+    constructor(props: ListviewSortProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: "",
+            findingWidget: true
+        };
+        this.updateSort = this.updateSort.bind(this);
+        dojoConnect.connect(props.mxform, "onNavigation", this, dojoLang.hitch(this, this.initSort));
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-listview-sort", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            createElement(ValidateConfigs, {
+                ...this.props as ListviewSortProps,
+                queryNode: this.state.targetNode,
+                targetGrid: this.state.targetGrid,
+                targetGridName: this.props.targetGridName,
+                validate: !this.state.findingWidget
+            }),
+            this.renderBar()
+        );
+    }
+
+    private renderBar(): ReactElement<DropdownProps> {
+        if (this.state.validationPassed) {
+            return createElement(Dropdown, {
+                ...this.props as CommonProps,
+                onDropdownChangeAction: this.updateSort,
+                style: parseStyle(this.props.style)
+            });
+        }
+
+        return null;
+    }
+
+    private initSort() {
```

Comment:
maybe find better name for this function. `connectListview` ?

Reply:
i have renamed it to `validate`